packages/engine/src/scene/components/SplineComponent.ts

```diff
@@ -23,33 +23,118 @@ All portions of the code written by the Ethereal Engine team are Copyright © 20
 Ethereal Engine. All Rights Reserved.
 */
 
-import { Vector3 } from 'three'
+import {
+  ArrowHelper,
+  BoxGeometry,
+  BufferAttribute,
+  BufferGeometry,
+  CatmullRomCurve3,
+  Line,
+  LineBasicMaterial,
+  Mesh,
+  MeshLambertMaterial,
+  Quaternion,
+  Vector3
+} from 'three'
 
-import { defineComponent } from '../../ecs/functions/ComponentFunctions'
+import { useEffect } from 'react'
 
-export type SplineComponentType = {
-  splinePositions: Vector3[]
+import { defineComponent, useComponent } from '../../ecs/functions/ComponentFunctions'
+import { useEntityContext } from '../../ecs/functions/EntityFunctions'
+import { ObjectLayers } from '../constants/ObjectLayers'
+import { setObjectLayers } from '../functions/setObjectLayers'
+import { addObjectToGroup, removeObjectFromGroup } from './GroupComponent'
+
+// an idea of a bundle that packages up all the related concerns on a spline point
+// @todo there may need to be a concept of an engine mode versus editor mode where no gizmos are manufactured
+
+const ARC_SEGMENTS = 200
```

Comment:
segments count should be configurable on the spline, rather than hard-coded to 200